File: API/Entities/LoanOffer.cs
using System;

namespace API.Entities;

public class LoanOffer
{
   public int Id { get; set; }
    public int LenderId { get; set; }
    public User Lender {get; set; }
    public decimal PrincipalAmount { get; set; }
    public decimal InterestRate { get; set; }
    public int DurationInYears { get; set; }
    public decimal MonthlyRepayment {get;set;}
    public decimal TotalRepayment {get;set;}
    public bool IsActive {get; set; }
    public bool HasApplied {get;set;}
    public DateTime CreatedAt {get; set;} = DateTime.UtcNow;
    public DateTime UpdatedAt {get; set;} = DateTime.UtcNow;



}

File: client/src/app/_models/loan-offer.ts
import { Lender } from "./lender";

export interface LoanOffer {
  id: number;
  lenderId: number;
  lender: Lender;
  principalAmount: number; // Using number for decimal in TypeScript
  interestRate: number;
  durationInYears: number;
  monthlyRepayment: number;
  totalRepayment: number;
  isActive: boolean;
  hasApplied: boolean; 
  createdAt: Date;
  updatedAt: Date;
}